Sources/Plasma/FeatureLib/pfDXPipeline/plDXRenderTargetRef.h

```diff
@@ -78,8 +78,19 @@ class plDXRenderTargetRef: public plDXTextureRef
 
         virtual void    SetOwner( plRenderTarget *targ ) { fOwner = (plBitmap *)targ; }
 
-        void    SetTexture( IDirect3DSurface9 *surface, IDirect3DSurface9 *depth );
-        void    SetTexture( IDirect3DTexture9 *surface, IDirect3DSurface9 *depth );
+        template <typename ColorType, typename DepthType>
+        void SetTexture(ColorType color, DepthType depth)
+        {
+            if constexpr (std::is_same<ColorType, IDirect3DSurface9*>::value) {
+                fD3DColorSurface = color;
+                fD3DTexture = nullptr;
+            }
+            else if constexpr (std::is_same<ColorType, IDirect3DTexture9*>::value) {
+                fD3DTexture = color;
+                fD3DColorSurface = nullptr;
+            }
+            fD3DDepthSurface = depth;
+        }
```

Comment:
Is there a reason DepthType is part of the template if there is only one possible DepthType? I'm still thinking over the use of a template here. The double functions were weird, but it's was also nice to have absolute types the arguments needed to be. Even if I had to read the functions a few times.

Reply:
simplified the template in the latest commit